plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/manager/VmwareStorageManagerImpl.java

```diff
@@ -1464,4 +1440,45 @@ private String deleteDir(String dir) {
     private static String getVolumeRelativeDirInSecStroage(long volumeId) {
         return "volumes/" + volumeId;
     }
+
+    /**
+     * This method is to check if a given TaskInfo Object is valid( and has name and entity name assigned).It return true if TaskInfo Object is valid and false otherwise.
+     *
+     * @param TaskInfo
+     *        info
+     * @return boolean(true or false)
+     **/
+    private boolean isvalidTaskInfoObj(TaskInfo info){
+        return !(info == null || info.getEntityName() == null || info.getName() == null);
+    }
+
+    private boolean waitForRunningTaskOnVM(String vmName, String taskName, VmwareContext context) throws Exception {
+        try {
+            ManagedObjectReference taskmgr = context.getServiceContent().getTaskManager();
+            List<ManagedObjectReference> tasks = context.getVimClient().getDynamicProperty(taskmgr, "recentTask");
+
+            for (ManagedObjectReference taskMor : tasks) {
+                TaskInfo info = (TaskInfo)(context.getVimClient().getDynamicProperty(taskMor, "info"));
+
+                if (!isvalidTaskInfoObj(info)) {
```

Comment:
I noticed you are only suggesting ;), no need to say that, when we review a PR we suggest changes to improve it, which is normal.

I am going to ask again, Do you mean remove the “isvalidTaskInfoObj” and add that condition to the (if)?
That would not reduce the code complexity, in the worst case scenario the complexity would be the same. At the line you commented, when we use that “isvalidTaskInfoObj” to mask the complexity on the logic to check if a task, we can remove some complexity from one method to another; doing that, we facilitate code documentation and unit tests writing.

For instance, if you want to suggest something to reduce code complexity or the so called cyclomatic complexity, you should take a look at lines 1467 and 1468, there you have a lot of room for improvements.